core/src/main/scala/eu/shiftforward/apso/json/Implicits.scala

```diff
@@ -83,6 +85,58 @@ object Implicits {
     }
   }
 
+  final implicit class ApsoJsonObject(val json: Json) extends AnyVal {
+    /**
+     * Returns a set of keys of this object where nested keys are separated by a separator character.
+     *
+     * Eg. {"a":1,"b":{"c":2},"d":null}.flattenedKeySet(".", ignoreNull = true) = Set("a","b.c")
+     *
+     * @param separator character separator to use
+     * @param ignoreNull if set, fields with a null value are ignored
+     * @return flattened key set
+     */
+    def flattenedKeySet(separator: String = ".", ignoreNull: Boolean = true): Set[String] = {
+      val fields = json.as[JsonObject].toOption.get.toMap.toSet
```

Comment:
This will throw if `json` can't be converted to `JsonObject`. I think we can avoid it by either making `json` be a `JsonObject`, which might make it more difficult to use this, or returning an empty `Set[String]` in case `json` is not an object.

Reply:
I've changed it on 10df9c1f44a85ada171c60aac82d3b5fb404b961 to return empty `Set[String]` on conversion error